### PR TITLE
Fixing RDTSC on Xeon/Phi

### LINKED ISSUE #1616: cmake problem resulting in RDTSC not working correctly for Xeon Phi creates very strange results for duration counters
The counters that use time calculations such as task_overhead and duration counters give results of zero with the following cmake command, I get HPX_WITH_RDTSC_UNINITIALIZED=ON

```
cmake -DCMAKE_INSTALL_PREFIX=~/build/hpx_buildmic_idle \
      -DCMAKE_BUILD_TYPE=Release \
      -DCMAKE_TOOLCHAIN_FILE=$HOME/hpx/cmake/toolchains/XeonPhi.cmake \
      -DBOOST_ROOT=~/boost/boost_1_55_0_mic \
      -DHWLOC_ROOT=/opt/hwloc/1.7-k1om-release \
      -DHPX_WITH_MALLOC=tbbmalloc \
      -DTBBMALLOC_ROOT=/opt/intel/tbb/ \
      -DHPX_WITH_THREAD_GUARD_PAGE=OFF \
      -DHPX_WITH_EXAMPLES=ON \
      -DHPX_WITH_TESTS=ON \
      -DHPX_WITH_THREAD_IDLE_RATES=ON \
       ~/hpx > >(tee cmakeout) 2> >(tee cmakeerr)
```

I tried adding -DHPX_WITH_RDTSC=ON nothing changed.

Here is the CMakeCache.txt file

### PR DESCRIPTION
**Pull request by hkaiser**
This fixes #1616

### CODE REVIEW

**Comment by hkaiser** [comment]
As mentioned in #1616, this patch fixes the issue.

**Comment by pagrubel** [comment]
This works fine for me on the phi

File: cmake/HPX_AddConfigTest.cmake [hunk 1]
```diff
@@ -57,14 +57,14 @@ macro(add_hpx_config_test variable)
             "-DCOMPILE_DEFINITIONS=${CONFIG_TEST_COMPILE_DEFINITIONS}"
           RUN_OUTPUT_VARIABLE ${variable}_OUTPUT
           ARGS ${${variable}_ARGS})
-          if(${variable}_COMPILE_RESULT AND NOT ${variable}_RUN_RESULT)
-            set(${variable}_RESULT TRUE)
-          else()
-            set(${variable}_RESULT FALSE)
```
Comment (sithhell) [inline cmake/HPX_AddConfigTest.cmake]: What happens if the result hasn't been run? is `${variable}_RUN_RESULT` set to `FALSE` or `TRUE`?
Are all config tests automatically set to `FALSE` if we are cross compiling now?

Comment (hkaiser) [inline cmake/HPX_AddConfigTest.cmake]: Good point. I'm not married to this particular change, will revert.

Comment (hkaiser) [inline cmake/HPX_AddConfigTest.cmake]: Actually, this is just a formatting change (indentation) which in the diff looks like a real change. So there is no need to revert anything.

**Comment by pagrubel** [comment]
Good

**Pull request merged**